cmip6_downscaling/analysis/metrics.py

```diff
@@ -59,3 +60,154 @@ def calc(obj, compute=False):
+
+
+def is_wet_day(ds: xr.Dataset, threshold: float = 0.01):
+    """[summary]
+
+    Parameters
+    ----------
+    ds : xr.Dataset
+        [description]
+    threshold : float, optional
+        the threshold for precipitation (mm/day) above which a day is consisdered wet, by default 0.01
+    """
+    return ds > threshold
+
+
+def metric_calc(ds, metric, dim='time', skipna=False):
+    if metric == 'mean':
+        return ds.mean(dim='time', skipna=skipna)
+    elif metric == 'median':
+        return ds.median(dim='time', skipna=skipna)
+    elif metric == 'std':
+        return ds.std(dim='time', skipna=skipna)
+    elif 'percentile' in metric:
+        # parse the percentile
+        percentile = float(metric.split('percentile')[1]) / 100
+        ds = ds.chunk({'time': -1})
+        return ds.quantile(percentile, dim='time', skipna=skipna)
+    else:
+        raise NotImplementedError
+
+
+def wet_day_amount(ds: xr.Dataset, method: str = 'mean', threshold: float = 0.01):
+    """Extract days that received precipitation above a given threshold
+    and then do statistics on them.
+
+    Parameters
+    ----------
+    ds : xr.Dataset
+        Dataset including precipitation at a daily temporal scale.
+    method : str, optional
+        statistic you want to assess of wet days, by default 'mean'
+    threshold : float, optional
+        the threshold for precipitation (mm/day) above which a day is consisdered wet, by default 0.01
+
+    Returns
+    -------
+    xr.Dataset
+
+
+    Raises
+    ------
+    NotImplementedError
+        [description]
+    """
+    assert 'pr' in ds, 'Precipitation not in dataset'
+    wet_day = ds.where(is_wet_day(ds, threshold=threshold))
+    return metric_calc(wet_day, method, skipna=True)
+
+
+def probability_two_consecutive_days(
+    ds: xr.Dataset, kind_of_day: str = 'wet', threshold: float = 0.01
+):
+    """Contitional probability of a day being the same as the day before.
+    So, given the fact that it was wet yesterday, what is the probability it will be wet today.
+    Or, if it was dry yesterday, what is probability it will be dry today.
+
+    Parameters
+    ----------
+    ds : xr.Dataset
+        dataset including daily precipitation
+    kind_of_day : str, optional
+        The kind of day you are interested in, by default 'wet'
+    threshold : float, optional
+        Threshold to determine whether day is wet or dry, by default 0.01
+
+    Returns
+    -------
+    xr.Dataset
+        Probability of a day being the same as the day before it
+    """
+    if kind_of_day == 'wet':
+        valid_days = is_wet_day(ds, threshold=threshold)
+    elif kind_of_day == 'dry':
+        valid_days = ~is_wet_day(ds, threshold=threshold)
+    yesterday_valid = valid_days.roll({'time': 1})
+    return valid_days.where(yesterday_valid).mean(dim='time', skipna=True)
+
+
+def probability_wet(ds: xr.Dataset, threshold: float = 0.01):
```

Comment:
```suggestion
def probability_wet(ds: xr.Dataset, threshold: float = 0.01) -> xr.Dataset:
```

Also add returns to docstring

Reply:
done